### PR TITLE
Removes barriers in conversions from a Qiskit circuit.

### LINKED ISSUE #350: Barriers in Qiskit circuits cause conversion errors (e.g., when folding)
In most cases I think barriers could be removed without affecting the circuit execution -- should we ignore/remove barriers when converting to a Cirq circuit?

### PR DESCRIPTION
**Pull request by rmlarose**
Fixes #350.

### CODE REVIEW

File: mitiq/mitiq_qiskit/conversions.py
Comment (andreamari) [inline mitiq/mitiq_qiskit/conversions.py:92]: Do we need to `_remove_barriers` in the body of "from_qasm"? Isn't `_remove_barriers` inside `from_qiskit` enough?

File: mitiq/mitiq_qiskit/conversions.py
Comment (andreamari) [inline mitiq/mitiq_qiskit/conversions.py:79]: If we decide to keep  `_remove_barriers` in the body of `from_qasm`, maybe we can remove  `_remove_barriers` from the body of `from_qiskit` (since `from_qasm` is called one line below).
```suggestion
```

**Review by andreamari — COMMENTED** [review]
Thanks Ryan! I left two minor (mutually exclusive) comments.

**Comment by rmlarose** [comment]
Oh yeah, oops. Only one call is necessary, thanks @andreamari.

**Comment by rmlarose** [comment]
@karalekas I'll wait for your review before merging. Feel free to merge if it looks good.

**Review by andreamari — APPROVED** [review]
Thanks LGTM

**Review by karalekas — APPROVED** [review]
LGTM!

**Pull request merged**
